woodwork/schema.py

```diff
@@ -327,6 +327,43 @@ def set_time_index(self, new_time_index):
             _check_time_index(self.columns.keys(), new_time_index, self.logical_types.get(new_time_index))
             self._set_time_index_tags(new_time_index)
 
+    def rename(self, columns):
+        """Renames columns in a Schema
+
+        Args:
+            columns (dict[str -> str]): A dictionary mapping columns whose names
+                we'd like to change to the name to which we'd like to change them.
+
+        Returns:
+            woodwork.Schema: Schema with the specified columns renamed.
+
+        Note:
+            Index and time index columns cannot be renamed.
+        """
+        if not isinstance(columns, dict):
+            raise TypeError('columns must be a dictionary')
+
+        for old_name, new_name in columns.items():
+            if old_name not in self.columns:
+                raise KeyError(f"Column to rename must be present. {old_name} cannot be found.")
```

Comment:
Raise a custom woodwork exceptions for these (which are subclasses properly `class ColumnNotPresent(KeyError)`)?

Reply:
I wonder if this should subclass LookupError? 

We have a couple of other exceptions that we throw when a column that a user asks for isn't present (rename, drop, pop, reset_semantic_tags, _check_logical_types) and we use LookupError for most of those, but not all of them. Drop uses a ValueError, and rename is using a KeyError. Ideally, I think we'd be able to use the same exception in all of those cases.

Reply:
Looking at the Python [docs](https://docs.python.org/3/library/exceptions.html#LookupError), `LookupError` is actually the base of KeyError. Since KeyError is specifically for a dictionaries, let's use LookupError to specify the column was not found in the DataFrame. 

### exception LookupError
> The base class for the exceptions that are raised when a key or index used on a mapping or sequence is invalid: IndexError, KeyError. This can be raised directly by codecs.lookup().
> 

###  exception KeyError
> Raised when a mapping (dictionary) key is not found in the set of existing keys.

Reply:
Added a `ColumnNotPresentError` that's used by `rename`. If we want to use this error in the other methods that are throwing similar errors, then we should probably do that as a part of the improvements epic.

Reply:
Added: https://github.com/alteryx/woodwork/issues/649

Reply:
Thanks!